### ISSUE TITLE
EnumFactory.getEnum(Class<?>) returns null

### ISSUE THREAD

**Issue by simonedavico** [INRIA/spoon#563]
Simple test case I was trying while playing around with `EnumFactory`:

```java
import java.time.DayOfWeek;

import spoon.reflect.declaration.*;
import spoon.processing.AbstractProcessor;

public class ClassProcessor extends AbstractProcessor<CtClass> {
    
    public void process(CtClass element) {
          CtEnum<DayOfWeek> dayOfWeekEnum = getFactory().Enum().getEnum(DayOfWeek.class);
          dayOfWeekEnum.getEnumValue("MONDAY"); 
    }

}
```

This processor will throw a `NullPointerException`, since for some reason `dayOfWeekEnum` will be null.
Am I doing something wrong, or is this a bug in spoon?

I am using `5.1.0` from Maven.

**Comment by monperrus** [comment]
It's probably because DayOfWeek.java is not in the spoon source folder.

The feature of reflecting the structural part of runtime classes as Spoon objects is very interesting IMHO

**Comment by simonedavico** [comment]
Thank you for the answer @monperrus. From what I understand, it means that I can't get an enum value for an arbitrary enum using spoon? Or maybe there is an alternative way?

**Comment by GerardPaligot** [comment]
Hi @simonedavico,

- What is the version 2.18.4.Final of Spoon? I don't know this version.
- `DayOfWeek` is a top level enum?
- `DayOfWeek` is specified in the input source of the Spoon launcher?

**Comment by simonedavico** [comment]
Hi @GerardPaligot,

- Sorry about the version number, I copied the wrong one from my POM; I am using the latest version, `5.1.0`
- `DayOfWeek` is an enum in `java.time`;
- I didn't specify it as input source to the Spoon launcher. This is how I am launching Spoon:

```java
public class SpoonTest {

    public static void main(String[] args) {

        Path outputDirectory = Paths.get("./src/main/resources");
        Launcher spoonLauncher = new spoon.Launcher();
        spoonLauncher.setSourceOutputDirectory(outputDirectory.toFile());
        spoonLauncher.addProcessor(new ClassProcessor());
        spoonLauncher.addInputResource("./testSpoon/Test.java");
        System.out.println("Before run");
        spoonLauncher.run();

    }
}
```

As you can see, I am only adding as source the class I want to modify (`Test.java`). My final aim is to add an annotation to the class which as an enum value as annotation value, such as

```
@DefaultDay(day = DayOfWeek.MONDAY)
public class Test {
    //...
}
```

**Comment by GerardPaligot** [comment]
Ok, I understand your issue.

Spoon doesn't build java classes in models. If you want to react with java classes, you must create `CtTypeReference` for your needs.

For example, if you want all classes annotated with an annotation named `DefaultDay` and with a field value, you can write something like this:

```java
public class ClassProcessor extends AbstractProcessor<CtClass> {
	@Override
	public boolean isToBeProcessed(CtClass candidate) {
		final CtAnnotation<DefaultDay> annotation = candidate.getAnnotation(getFactory().Type().createReference(DefaultDay.class));
		final CtExpression day = annotation.getValue("day");
		return day instanceof CtFieldRead;
	}

	public void process(CtClass element) {
		// Make what do you want here.
	}
}
```

**Comment by simonedavico** [comment]
@GerardPaligot thank you for your response (that I completely missed, sorry about the late reply).

What I am looking for is not really checking whether my class has an annotation `@DefaultDay`; I would like to ADD the annotation to the class! 

So, for example, I would like to go from

```java
public class Test {
    //body of the class
}
```

to 

```java
@DefaultDay(day = DayOfWeek.MONDAY)
public class Test {
    //body of the class
}
```

Is there a way I can achieve this with Spoon?

**Comment by tdurieux** [comment]
Hi,

In order to create a what you need, you have to do something like this:
```java
CtFieldAccess<?> fieldRead = factory.Core().createFieldRead();
CtTypeReference<DayOfWeek> enumReference = factory.Type().createReference(DayOfWeek.class);
CtFieldReference fieldReference = factory.Field().createReference(enumReference, enumReference, DayOfWeek.MONDAY.name());
fieldReference.setStatic(true);
fieldRead.setVariable(fieldReference);
CtAnnotation<?> ctAnnotation = element.getAnnotation(getFactory().Type().createReference(DefaultDay.class));
ctAnnotation.addValue("day", fieldRead);
```

**Comment by simonedavico** [comment]
@tdurieux thank you, it works as expected!

**Comment by GerardPaligot** [comment]
Fixed by #569
